hs_discover/views.py

```diff
@@ -1,57 +1,224 @@
+import datetime
 import json
+import time
+import logging
+from collections import namedtuple
 
+from django.core.paginator import Paginator
+from django.http import JsonResponse
 from django.shortcuts import render
 from django.views.generic import TemplateView
 from haystack.query import SearchQuerySet
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+logger = logging.getLogger(__name__)
+
+DateRange = namedtuple('DateRange', ['start', 'end'])
 
 
-# TODO error handling, validation, analytics
 class SearchView(TemplateView):
 
     def get(self, request, *args, **kwargs):
-        from django.template.defaultfilters import date, time
-        q = request.GET.get('q') if request.GET.get('q') else ""
+        return render(request, 'hs_discover/index.html', {})
+
+
+class SearchAPI(APIView):
+
+    def __init__(self, **kwargs):
+        super(SearchAPI, self).__init__(**kwargs)
+        self.filterlimit = 20
+        self.perpage = 35
```

Comment:
It's worth considering to make these defaults configurable in settings.  I think it's reasonable to at least make perpage configurable in the request

Reply:
If such fine grained items were in settings/local_settings that would be too much. The class is a good place to store these. As far as making the API more generically useable, let's hold off on discussions of _hsapi and swagger and if an external client would even use this (which I think it should down the road). The icon sizes have made rows thinner, so I'm updating to 40 items. Also future discussions should consider the user profile storing per page preferences among other HS prefs.